java/statestore/src/test/java/sleeper/statestore/transactionlog/TransactionLogSnapshotDeleterIT.java

```diff
@@ -0,0 +1,250 @@
+import static sleeper.configuration.properties.instance.CommonProperty.FILE_SYSTEM;
+import static sleeper.configuration.properties.table.TablePropertiesTestHelper.createTestTableProperties;
+import static sleeper.configuration.properties.table.TableProperty.STATESTORE_CLASSNAME;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_ID;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.configuration.properties.table.TableProperty.TRANSACTION_LOG_SNAPSHOT_EXPIRY_IN_DAYS;
+import static sleeper.core.schema.SchemaTestHelper.schemaWithKey;
+
+public class TransactionLogSnapshotDeleterIT extends TransactionLogStateStoreTestBase {
+    @TempDir
+    private java.nio.file.Path tempDir;
+    private final Schema schema = schemaWithKey("key", new LongType());
+    private FileSystem fs;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        instanceProperties.set(FILE_SYSTEM, "file://");
+        instanceProperties.set(DATA_BUCKET, tempDir.toString());
+        fs = FileSystem.get(configuration);
+    }
+
+    @Test
+    void shouldDeleteOldSnapshotsForOneTable() throws Exception {
+        // Given
+        TableProperties table = createTable("test-table-id-1", "test-table-1");
+        table.setNumber(TRANSACTION_LOG_SNAPSHOT_EXPIRY_IN_DAYS, 1);
+        StateStore stateStore = createStateStore(table);
+        PartitionsBuilder partitionsBuilder = new PartitionsBuilder(schema).rootFirst("root");
+        stateStore.initialise(partitionsBuilder.buildList());
+        FileReferenceFactory factory = FileReferenceFactory.from(stateStore);
+        stateStore.addFile(factory.rootFile("test1.parquet", 123L));
+        createSnapshotsAt(table, Instant.parse("2024-04-25T11:24:00Z"));
+
+        stateStore.clearFileData();
+        stateStore.initialise(partitionsBuilder.splitToNewChildren("root", "L", "R", 123L).buildList());
+        factory = FileReferenceFactory.from(stateStore);
+        stateStore.addFile(factory.rootFile("test3.parquet", 789L));
+        createSnapshotsAt(table, Instant.parse("2024-04-27T11:24:00Z"));
+
+        // When
+        deleteSnapshotsAt(table, Instant.parse("2024-04-27T11:24:00Z"));
+
+        // Then
+        assertThat(snapshotStore(table).getLatestSnapshots())
+                .isEqualTo(new LatestSnapshots(
+                        filesSnapshot(table, 3),
+                        partitionsSnapshot(table, 2)));
+        assertThat(snapshotStore(table).getFilesSnapshots())
+                .containsExactly(filesSnapshot(table, 3));
+        assertThat(snapshotStore(table).getPartitionsSnapshots())
+                .containsExactly(partitionsSnapshot(table, 2));
+        assertThat(filesSnapshotFileExists(table, 3)).isTrue();
+        assertThat(filesSnapshotFileExists(table, 1)).isFalse();
+        assertThat(partitionsSnapshotFileExists(table, 2)).isTrue();
+        assertThat(partitionsSnapshotFileExists(table, 1)).isFalse();
```

Comment:
Maybe it would be worth asserting the full contents of the temp directory, rather than 4 separate assertions?